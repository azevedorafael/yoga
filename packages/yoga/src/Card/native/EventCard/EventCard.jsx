import React from 'react';
import styled from 'styled-components';
import { oneOf, shape, string } from 'prop-types';
import { View } from 'react-native';
import { Clock } from '@gympass/yoga-icons';

import Card from '../Card';
import Text from '../../../Text';

const Event = styled(Card)`
  flex-direction: row;

  width: 280px;
  height: 104px;
  padding: 0;
`;

const DateInfo = styled(View)`
  ${({
    variant,
    theme: {
      yoga: {
        components: {
          card: { event },
        },
        colors: { [variant]: color = {} },
      },
    },
  }) => `
    justify-content: center;
    align-items: center;

    width: 56px;

    border-top-left-radius: ${event.date.radius}px;
    border-bottom-left-radius: ${event.date.radius}px;
    background-color: ${color[2]};
  `}
`;

const EventInfo = styled(View)`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => `
    padding: ${event.info.padding}px;
  `}
`;

const Weekday = styled(Text.Small)`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => `
    padding-bottom: ${event.date.dayOfWeek.padding.bottom}px;

    font-weight: ${event.date.dayOfWeek.fontWeight};
  `}
`;

const Month = styled(Text.Small)`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => `
    font-weight: ${event.date.month.fontWeight};
  `}
`;

const Name = styled(Text.H4)`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => `
    padding-bottom: ${event.info.name.padding.bottom}px;

    font-weight: ${event.info.name.fontWeight};
  `}
`;

const Place = styled(Text.Small)`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => `
    padding-bottom: ${event.info.place.padding.bottom}px;

    color: ${event.info.place.color};
  `}
`;

const EventCard = ({ event, date, variant, ...rest }) => (
  <Event {...rest}>
    <DateInfo variant={variant}>
      <Weekday inverted>{date.weekday}</Weekday>
      <Text.H3 inverted>{date.day}</Text.H3>
      <Month inverted>{date.month}</Month>
    </DateInfo>
    <EventInfo>
      <Name>{event.name}</Name>
      <Place>{event.place}</Place>
      <View style={{ flexDirection: 'row', alignItems: 'center' }}>
        <Clock style={{ marginRight: 5 }} />
        <Text.Tiny>{event.time}</Text.Tiny>
      </View>
    </EventInfo>
  </Event>
);

EventCard.propTypes = {
  /** event information: { name (string), place (string), time (string) } */
  event: shape({
    name: string,
    place: string,
    time: string,
  }).isRequired,
  /** date information: { day (string), weekday (string), month (string) } */
  date: shape({
    day: string,
    weekday: string,
    month: string,
  }).isRequired,
  /** style the card following the theme (primary, secondary, tertiary) */
  variant: oneOf(['primary', 'secondary', 'tertiary']),
};

EventCard.defaultProps = {
  variant: 'secondary',
};

EventCard.displayName = 'EventCard';

export default EventCard;
