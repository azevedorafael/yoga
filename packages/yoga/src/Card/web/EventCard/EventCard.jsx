import React from 'react';
import styled, { withTheme, css } from 'styled-components';
import { oneOf, shape, string } from 'prop-types';
import { Clock } from '@gympass/yoga-icons';

import Card from '../Card';
import Text from '../../../Text';

const Event = styled(Card)`
  display: flex;
  flex-direction: row;

  width: 100%;
  padding: 0;
`;

const EventInfo = styled.div`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => css`
    padding: ${event.info.padding}px;

    ${Text.H4} {
      padding-bottom: ${event.info.name.padding.bottom}px;

      font-weight: ${event.info.name.fontWeight};
    }

    ${Text.Small} {
      padding-bottom: ${event.info.place.padding.bottom}px;

      color: ${event.info.place.color};
    }
  `}
`;

const EventTime = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

const DateInfo = styled.div`
  ${({
    variant,
    theme: {
      yoga: {
        components: {
          card: { event },
        },
        colors: { [variant]: color },
      },
    },
  }) => `
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    flex-shrink: 0;

    width: 56px;

    border-top-left-radius: ${event.date.radius}px;
    border-bottom-left-radius: ${event.date.radius}px;
    background-color: ${color[2]};
  `}
`;

const Weekday = styled(Text.Small)`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => `
    padding-bottom: ${event.date.weekday.padding.bottom}px;

    font-weight: ${event.date.weekday.fontWeight};
  `}
`;

const Month = styled(Text.Small)`
  ${({
    theme: {
      yoga: {
        components: {
          card: { event },
        },
      },
    },
  }) => `
    font-weight: ${event.date.month.fontWeight};
  `}
`;

const EventCard = ({
  event,
  date,
  variant,
  theme: {
    yoga: {
      components: {
        card: {
          event: { icon },
        },
      },
    },
  },
  ...rest
}) => (
  <Event {...rest}>
    <DateInfo variant={variant}>
      <Weekday inverted>{date.weekday}</Weekday>
      <Text.H3 inverted>{date.day}</Text.H3>
      <Month inverted>{date.month}</Month>
    </DateInfo>
    <EventInfo>
      <Text.H4>{event.name}</Text.H4>
      <Text.Small>{event.place}</Text.Small>
      <EventTime>
        <Clock fill={icon.fill} style={{ marginRight: 5 }} />
        <Text.Tiny>{event.time}</Text.Tiny>
      </EventTime>
    </EventInfo>
  </Event>
);

EventCard.propTypes = {
  /** event information: { name (string), place (string), time (string) } */
  event: shape({
    name: string,
    place: string,
    time: string,
  }).isRequired,
  /** date information: { day (string), weekday (string), month (string) } */
  date: shape({
    day: string,
    weekday: string,
    month: string,
  }).isRequired,
  /** style the card following the theme (primary, secondary, tertiary) */
  variant: oneOf(['primary', 'secondary', 'tertiary']),
};

EventCard.defaultProps = {
  variant: 'secondary',
};

EventCard.displayName = 'EventCard';

export default withTheme(EventCard);
