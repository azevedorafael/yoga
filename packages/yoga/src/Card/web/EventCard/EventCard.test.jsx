import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from '../../..';
import EventCard from '.';

const event = {
  name: 'Yoga Class',
  place: 'Gympass',
  time: '19 am',
};

const date = {
  day: '19',
  weekday: 'thu',
  month: 'dec',
};

describe('<Card />', () => {
  describe('Snapshots', () => {
    it('should match snapshot with default EventCard', () => {
      const { container } = render(
        <ThemeProvider>
          <EventCard event={event} date={date} />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });
    it('should match snapshot with variant primary EventCard', () => {
      const { container } = render(
        <ThemeProvider>
          <EventCard event={event} date={date} variant="primary" />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });
    it('should match snapshot with variant secondary EventCard', () => {
      const { container } = render(
        <ThemeProvider>
          <EventCard event={event} date={date} variant="secondary" />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });
    it('should match snapshot with variant tertiary EventCard', () => {
      const { container } = render(
        <ThemeProvider>
          <EventCard event={event} date={date} variant="tertiary" />
        </ThemeProvider>,
      );
      expect(container).toMatchSnapshot();
    });
  });
});
