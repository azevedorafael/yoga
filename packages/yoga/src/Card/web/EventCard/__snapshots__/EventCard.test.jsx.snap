// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> Snapshots should match snapshot with default EventCard 1`] = `
.c5 {
  margin: 0;
  padding: 0;
  font-size: 20px;
  line-height: 24px;
  font-weight: 700;
  font-family: Open Sans;
  color: #41414A;
  color: #FFFFFF;
}

.c9 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 20px;
  font-weight: 700;
  font-family: Open Sans;
  color: #41414A;
}

.c10 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
}

.c12 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
}

.c0 {
  padding: 16px  16px  16px  16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  padding: 0;
}

.c7 {
  padding: 16px;
}

.c7 .c8 {
  padding-bottom: 8px;
  font-weight: 600;
}

.c7 .c3 {
  padding-bottom: 8px;
  color: #6B6B78;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 56px;
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  background-color: #8051E1;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
  color: #FFFFFF;
  padding-bottom: 8px;
  font-weight: 400;
}

.c6 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
  color: #FFFFFF;
  font-weight: 600;
}

<div>
  <section
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <p
        class="c3 c4"
      >
        thu
      </p>
      <h3
        class="c5"
      >
        19
      </h3>
      <p
        class="c3 c6"
      >
        dec
      </p>
    </div>
    <div
      class="c7"
    >
      <h4
        class="c8 c9"
      >
        Yoga Class
      </h4>
      <p
        class="c3 c10"
      >
        Gympass
      </p>
      <div
        class="c11"
      >
        <svg
          fill="#41414A"
          style="margin-right: 5px;"
        />
        <p
          class="c12"
        >
          19 am
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`<Card /> Snapshots should match snapshot with variant primary EventCard 1`] = `
.c5 {
  margin: 0;
  padding: 0;
  font-size: 20px;
  line-height: 24px;
  font-weight: 700;
  font-family: Open Sans;
  color: #41414A;
  color: #FFFFFF;
}

.c9 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 20px;
  font-weight: 700;
  font-family: Open Sans;
  color: #41414A;
}

.c10 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
}

.c12 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
}

.c0 {
  padding: 16px  16px  16px  16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  padding: 0;
}

.c7 {
  padding: 16px;
}

.c7 .c8 {
  padding-bottom: 8px;
  font-weight: 600;
}

.c7 .c3 {
  padding-bottom: 8px;
  color: #6B6B78;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 56px;
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  background-color: #F78D82;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
  color: #FFFFFF;
  padding-bottom: 8px;
  font-weight: 400;
}

.c6 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
  color: #FFFFFF;
  font-weight: 600;
}

<div>
  <section
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <p
        class="c3 c4"
      >
        thu
      </p>
      <h3
        class="c5"
      >
        19
      </h3>
      <p
        class="c3 c6"
      >
        dec
      </p>
    </div>
    <div
      class="c7"
    >
      <h4
        class="c8 c9"
      >
        Yoga Class
      </h4>
      <p
        class="c3 c10"
      >
        Gympass
      </p>
      <div
        class="c11"
      >
        <svg
          fill="#41414A"
          style="margin-right: 5px;"
        />
        <p
          class="c12"
        >
          19 am
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`<Card /> Snapshots should match snapshot with variant secondary EventCard 1`] = `
.c5 {
  margin: 0;
  padding: 0;
  font-size: 20px;
  line-height: 24px;
  font-weight: 700;
  font-family: Open Sans;
  color: #41414A;
  color: #FFFFFF;
}

.c9 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 20px;
  font-weight: 700;
  font-family: Open Sans;
  color: #41414A;
}

.c10 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
}

.c12 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
}

.c0 {
  padding: 16px  16px  16px  16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  padding: 0;
}

.c7 {
  padding: 16px;
}

.c7 .c8 {
  padding-bottom: 8px;
  font-weight: 600;
}

.c7 .c3 {
  padding-bottom: 8px;
  color: #6B6B78;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 56px;
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  background-color: #8051E1;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
  color: #FFFFFF;
  padding-bottom: 8px;
  font-weight: 400;
}

.c6 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
  color: #FFFFFF;
  font-weight: 600;
}

<div>
  <section
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <p
        class="c3 c4"
      >
        thu
      </p>
      <h3
        class="c5"
      >
        19
      </h3>
      <p
        class="c3 c6"
      >
        dec
      </p>
    </div>
    <div
      class="c7"
    >
      <h4
        class="c8 c9"
      >
        Yoga Class
      </h4>
      <p
        class="c3 c10"
      >
        Gympass
      </p>
      <div
        class="c11"
      >
        <svg
          fill="#41414A"
          style="margin-right: 5px;"
        />
        <p
          class="c12"
        >
          19 am
        </p>
      </div>
    </div>
  </section>
</div>
`;

exports[`<Card /> Snapshots should match snapshot with variant tertiary EventCard 1`] = `
.c5 {
  margin: 0;
  padding: 0;
  font-size: 20px;
  line-height: 24px;
  font-weight: 700;
  font-family: Open Sans;
  color: #41414A;
  color: #FFFFFF;
}

.c9 {
  margin: 0;
  padding: 0;
  font-size: 16px;
  line-height: 20px;
  font-weight: 700;
  font-family: Open Sans;
  color: #41414A;
}

.c10 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
}

.c12 {
  margin: 0;
  padding: 0;
  font-size: 12px;
  line-height: 16px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
}

.c0 {
  padding: 16px  16px  16px  16px;
  border-radius: 8px;
  background-color: #FFFFFF;
  box-shadow: 0 2px 6px rgba(152,152,166,0.25);
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  padding: 0;
}

.c7 {
  padding: 16px;
}

.c7 .c8 {
  padding-bottom: 8px;
  font-weight: 600;
}

.c7 .c3 {
  padding-bottom: 8px;
  color: #6B6B78;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 56px;
  border-top-left-radius: 8px;
  border-bottom-left-radius: 8px;
  background-color: #3FA99B;
}

.c4 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
  color: #FFFFFF;
  padding-bottom: 8px;
  font-weight: 400;
}

.c6 {
  margin: 0;
  padding: 0;
  font-size: 14px;
  line-height: 20px;
  font-weight: 400;
  font-family: Open Sans;
  color: #41414A;
  color: #FFFFFF;
  font-weight: 600;
}

<div>
  <section
    class="c0 c1"
  >
    <div
      class="c2"
    >
      <p
        class="c3 c4"
      >
        thu
      </p>
      <h3
        class="c5"
      >
        19
      </h3>
      <p
        class="c3 c6"
      >
        dec
      </p>
    </div>
    <div
      class="c7"
    >
      <h4
        class="c8 c9"
      >
        Yoga Class
      </h4>
      <p
        class="c3 c10"
      >
        Gympass
      </p>
      <div
        class="c11"
      >
        <svg
          fill="#41414A"
          style="margin-right: 5px;"
        />
        <p
          class="c12"
        >
          19 am
        </p>
      </div>
    </div>
  </section>
</div>
`;
